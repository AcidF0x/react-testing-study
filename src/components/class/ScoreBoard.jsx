import React from "react";

class ScoreBoard extends React.Component {
  render() {
    return (
      <div className="dice-score-board">
        <div className="text-center">
          <h3>총점수</h3>
          <div className="score">
            {this.props.totalScore}점
          </div>
        </div>
      </div>
    )
  }
}

export default ScoreBoard;