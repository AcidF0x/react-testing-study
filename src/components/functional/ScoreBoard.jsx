import React from "react";

const ScoreBoard = ({totalScore}) => {
  return (
    <div className="dice-score-board">
      <div className="text-center">
        <h3>총점수</h3>
        <div className="score">
          { totalScore }점
        </div>
      </div>
    </div>
  )
}

export default ScoreBoard;