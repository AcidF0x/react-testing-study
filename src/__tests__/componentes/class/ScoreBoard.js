import React from "react";
import ScoreBoard from "../../../components/class/ScoreBoard";
import { mount } from "enzyme";


describe('ScoreBoard 테스트', () => {
  it('렌더링을 할수있다', () => {
    // Given
    // When
    const wrapper = mount(<ScoreBoard/>);

    // Then
    expect(wrapper.text()).toContain('총점수')
  });
  it('스냅샷 테스트', () => {
    // Given
    const totalScore = 90;

    // When
    const wrapper = mount(<ScoreBoard totalScore={totalScore}/>);

    // Then
    expect(wrapper).toMatchSnapshot()
  });

  it('props로 전달 받은 내용을 렌더링 한다', () => {
    // Given
    const totalScore = 200;

    // When
    const wrapper = mount(<ScoreBoard totalScore={totalScore}/>);

    // Then
    const score = wrapper.find('.score')
    expect(score.text()).toBe(`${totalScore}점`)
  });
});
