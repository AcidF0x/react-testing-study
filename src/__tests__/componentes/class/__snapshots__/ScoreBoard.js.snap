// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard 테스트 스냅샷 테스트 1`] = `
<ScoreBoard
  totalScore={90}
>
  <div
    className="dice-score-board"
  >
    <div
      className="text-center"
    >
      <h3>
        총점수
      </h3>
      <div
        className="score"
      >
        90
        점
      </div>
    </div>
  </div>
</ScoreBoard>
`;
